conda/base/context.py

```diff
@@ -398,14 +386,6 @@ def __init__(self, search_path=None, argparse_args=None):
         super(Context, self).__init__(search_path=search_path, app_name=APP_NAME,
                                       argparse_args=argparse_args)
 
-        # Add plugin support
-        self._plugin_manager = get_plugin_manager()
-
-    @property
-    @functools.lru_cache(maxsize=None)  # FUTURE: Python 3.9+, replace w/ functools.cache
-    def plugin_manager(self):
-        return self._plugin_manager
```

Comment:
The reason we considered putting it into the context object, is that it's at the core of the conda startup process and is appropriate for loading user-specific configuration like the plugins.

Reply:
True, but I'm starting to suspect that we are misusing the context object, isn't it an anti-pattern to use a large stateful singleton? Moreover, our context object containing singletons feels like a no-no.

Keeping all of this within `conda.plugins` should also help with discoverability and is a step towards reorganizing conda into "topics" instead of "levels".